rasa/core/policies/rule_policy.py

```diff
@@ -231,9 +204,11 @@ def _create_form_unhappy_lookup_from_states(
             # even if there are two identical feature keys
             # their form will be the same
             # because of `active_form_...` feature
-            if active_form:
+            if active_form and active_form != SHOULD_NOT_BE_SET:
```

Comment:
I'm not sure, since it's hard to connect the number of memorised feature_keys and number of unhappy path. I'm afraid printing this number will confuse people